crates/ruff_cli/src/commands/format.rs

```diff
@@ -0,0 +1,104 @@
+#![allow(clippy::print_stderr)]
+
+use crate::args::{Arguments, Overrides};
+use crate::resolve::resolve;
+use crate::ExitStatus;
+use anyhow::bail;
+use colored::Colorize;
+use rayon::iter::{IntoParallelIterator, ParallelIterator};
+use ruff::resolver::python_files_in_path;
+use ruff::settings::types::{FilePattern, FilePatternSet};
+use ruff_formatter::LineWidth;
+use ruff_python_formatter::{format_module, FormatModuleError, PyFormatOptions};
+use std::io;
+use std::path::{Path, PathBuf};
+use thiserror::Error;
+use tracing::span;
+use tracing::Level;
+
+/// The inner errors are all flat, i.e. none of them has a source
+#[derive(Error, Debug)]
+enum FormatterIterationError {
+    #[error("Failed to traverse the inputs paths: {0}")]
+    Ignore(#[from] ignore::Error),
+    #[error("Failed to read {0}: {1}")]
+    Read(PathBuf, io::Error),
+    #[error("Failed to write {0}: {1}")]
+    Write(PathBuf, io::Error),
+    #[error("Failed to format {0}: {1}")]
+    FormatModule(PathBuf, FormatModuleError),
+}
+
+pub(crate) fn format(cli: &Arguments, overrides: &Overrides) -> anyhow::Result<ExitStatus> {
+    let mut pyproject_config = resolve(
+        cli.isolated,
+        cli.config.as_deref(),
+        overrides,
+        cli.stdin_filename.as_deref(),
+    )?;
+    // We don't want to format pyproject.toml
+    pyproject_config.settings.lib.include = FilePatternSet::try_from_vec(vec![
+        FilePattern::Builtin("*.py"),
+        FilePattern::Builtin("*.pyi"),
+    ])
+    .unwrap();
```

Comment:
I would suggest omitting this, and then in the `map` below, can we use `PySourceType::from(path)` and only run on Python files?

Reply:
`PySourceType` doesn't handle toml so it's now a strange mixture